app/models/katello/content_view_version.rb

```diff
@@ -348,6 +350,24 @@ def validate_destroyable!(skip_environment_check = false)
       true
     end
 
+    def before_promote_hook
+      run_callbacks :before_promote do
+        logger.debug "custom hook before_promote on #{name} will be executed if defined."
+        true
+      end
+    end
+
+    def after_promote_hook
+      run_callbacks :after_promote do
+        logger.debug "custom hook after_promote on #{name} will be executed if defined."
+        true
+      end
+    end
```

Comment:
replace both of these methods with:
```
    def promote_hook
      run_callbacks :promote do
        logger.debug "custom hook promote on #{name} will be executed if defined."
        true
      end
    end
```
and then you call:
`  ::Katello::ContentViewVersion.find(input[:id]).after_promote_hooks `  NOTE the plural hooks